### PR TITLE
feat(`forc doc`): Add `impl_trait` representation to `ItemBody`

### LINKED ISSUE #3972: Add handling of implementations to `forc doc`
Currently `ImplTrait`s are commented out and unhandled. I mention in a comment [here](https://github.com/FuelLabs/sway/blob/bdfae6cd478e28e0ca57fb0f89b5ecd9ebd16f12/forc-plugins/forc-doc/src/descriptor.rs#L205) that we should find a way to link implementations to their respective documents. This may mean either a hacky solution or update the AST to provide the implementations on said types.

### PR DESCRIPTION
**Pull request by eureka-cpu**
## Description
Adds a trait implementation section to item declarations & fixes CSS for required methods on traits.

https://user-images.githubusercontent.com/57543709/236067329-11bb66bd-b7b9-4a46-81d6-520d0e2ece83.mov

## Checklist

- [x] I have linked to any relevant issues. Closes #3972 
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation where relevant (API docs, the reference, and the Sway book).
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added (or requested a maintainer to add) the necessary `Breaking*` or `New Feature` labels where relevant.
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [x] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

File: forc-plugins/forc-doc/src/doc/mod.rs [hunk 1]
```diff
@@ -63,17 +63,46 @@ impl Document {
...
-                if let Descriptor::Documentable(doc) = desc {
-                    docs.push(doc)
+                    if let Descriptor::Documentable(doc) = desc {
+                        docs.push(doc)
+                    }
+                }
+            }
+        }
+        // match for the spans to add the impl_traits to their corresponding doc
+        if dbg!(!impl_traits.is_empty()) {
```
[the remark this answers is not in the document]
Comment (eureka-cpu) [reply forc-plugins/forc-doc/src/doc/mod.rs:86]: Hey @tritao this is still a work in progress

**Review by sdankel — DISMISSED** [review]
LGTM! It would be nice to add a screenshot to the description, and I see there are a few failing tests after updating the branch.

**Comment by eureka-cpu** [comment]
> LGTM! It would be nice to add a screenshot to the description, and I see there are a few failing tests after updating the branch.

Hey thanks, yeah I'll revert this to a draft since it isn't finished and I'm OOO

File: forc-plugins/forc-doc/src/render/item/context.rs [hunk 2]
```diff
@@ -309,23 +346,212 @@ impl ItemContext {
...
+                write!(
+                    fn_sig,
+                    "{} {},",
+                    param.name.as_str(),
+                    param.type_argument.span.as_str()
+                )?;
+            }
+        }
+        write!(fn_sig, ") -> {}", method.return_type.span.as_str())?;
+        let multiline = fn_sig.chars().count() >= 60;
```
Comment (sdankel) [inline forc-plugins/forc-doc/src/render/item/context.rs:476]: nit: move this to a helper function since we do this in two places

**Review by sdankel — APPROVED** [review]

**Review by mitchmindtree — APPROVED** [review]
Nice one, looking good!

File: forc-plugins/forc-doc/src/render/item/context.rs (hunk 2 above)
Comment (eureka-cpu) [reply forc-plugins/forc-doc/src/render/item/context.rs:476]: Right, I would have if I could have but the types are actually different even though they look the same and their fields don't always match up so I couldn't use a generic implementation. If you know of a way to go about this I don't mind making a follow up.

**Pull request merged**